api/v1alpha1/databasesecretengineconfig_types.go

```diff
@@ -150,9 +117,8 @@ func (r *DatabaseSecretEngineConfig) setInternalCredentials(context context.Cont
 		return nil
 	}
 	if r.Spec.RootCredentials.VaultSecret != nil {
-		secret, err := vaultClient.Logical().Read(string(r.Spec.RootCredentials.VaultSecret.Path))
+		secret, err := GetVaultKV2Secret(string(r.Spec.RootCredentials.VaultSecret.Path), context)
 		if err != nil {
-			log.Error(err, "unable to retrieve vault secret", "instance", r)
```

Comment:
I like to print error every time event if there is nothing to add...

Reply:
Error is printed by ReadSecret method, it will a duplication of the error if I add it back